processor/labelsprocessor/labels_processor.go

```diff
@@ -0,0 +1,124 @@
+	keys := make(map[string]bool)
+	for _, elem := range cfg.Labels {
+		_, value := keys[elem.Key]
+		if value {
+			return fmt.Errorf("Labels Processor configuration contains duplicate keys")
+		}
+		keys[elem.Key] = true
+	}
+
+	return nil
+}
+
+func (lp *labelMetricProcessor) ProcessMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+
+	otlpMetrics := pdata.MetricsToOtlp(md)
+
+	for _, otlpMetric := range otlpMetrics {
+		for _, instrMetric := range otlpMetric.GetInstrumentationLibraryMetrics() {
+			for _, metric := range instrMetric.GetMetrics() {
+
+				// Multiple types of Data Points exists, and each of them must be handled differently
+				if metric.GetIntSum() != nil {
+					intDataPoints := metric.GetIntSum().GetDataPoints()
+					handleIntDataPoints(intDataPoints, lp)
+				} else if metric.GetIntGauge() != nil {
+					intDataPoints := metric.GetIntGauge().GetDataPoints()
+					handleIntDataPoints(intDataPoints, lp)
+				} else if metric.GetDoubleGauge() != nil {
+					doubleDataPoints := metric.GetDoubleGauge().GetDataPoints()
+					handleDoubleDataPoints(doubleDataPoints, lp)
+				} else if metric.GetDoubleSum() != nil {
+					doubleDataPoints := metric.GetDoubleSum().GetDataPoints()
+					handleDoubleDataPoints(doubleDataPoints, lp)
+				} else if metric.GetIntHistogram() != nil {
+					intHistogramDataPoints := metric.GetIntHistogram().GetDataPoints()
+					handleIntHistogramDataPoints(intHistogramDataPoints, lp)
+				} else if metric.GetDoubleHistogram() != nil {
+					doubleHistogramDataPoints := metric.GetDoubleHistogram().GetDataPoints()
+					handleDoubleHistogramDataPoints(doubleHistogramDataPoints, lp)
+				}
+
+			}
+		}
+	}
+
+	return md, nil
+}
+
+func handleIntDataPoints(intDataPoints []*v1.IntDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range intDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+func handleDoubleDataPoints(doubleDataPoints []*v1.DoubleDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range doubleDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+func handleIntHistogramDataPoints(intHistogramDataPoints []*v1.IntHistogramDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range intHistogramDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+func handleDoubleHistogramDataPoints(doubleHistogramDataPoints []*v1.DoubleHistogramDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range doubleHistogramDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+// This processor will always by default update existing label values. Also assumes duplicate labels do not already exist in the metric
```

Comment:
nit: while this comment is good, it explains what the function is doing not "how", but consider if you can use better name for the function such that you don't need to have a comment for it. 

`DeDuplicateAndAppend` is a bit misleading here because this function is not really doing "deduplicate"; it is either overwriting or append. So maybe a term like "Upsert" or "UpdateOrInsert" would be a better term to use.

Reply:
now that I think about it, I agree the function name is not very representative of what it is actually doing, I will change it to "upsert" as that is a term used in other parts of the collector as well